### PR TITLE
feat: handle circular refs

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:

Handle circular refs:
- use new logic from https://github.com/asyncapi/parser-js/pull/391 PR
- make fallback for older solution

### CODE REVIEW

File: library/src/components/Schema.tsx [hunk 1]
```diff
@@ -47,25 +47,38 @@ export const Schema: React.FunctionComponent<Props> = ({
...
   let uid = schema.uid();
 
-  /**
-   * checking uid for circular items
-   * after fixing https://github.com/asyncapi/parser-js/issues/293 statement should be removed
-   * `x-parser-circular` extension is added to every schema which has circular `items` field,
-   * so we must check that `items` is schema (not array of schemas) and infer UID of schema to display which schema is circular (by the name of schema)
-   */
-  if (
+  // ane logic and fallback for older logic for circular references
```
Comment (jonaslagoni) [inline library/src/components/Schema.tsx:60]: 🤔

File: playground/src/specs/circular.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
+export const circular = `
```
Comment (jonaslagoni) [inline playground/src/specs/circular.ts:1]: Where is this used?

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Well, tests, I can't know if your feature works as expected 🤷 

Small comments.

File: library/src/components/Schema.tsx (hunk 1 above)
Comment (magicmatatjahu) [reply library/src/components/Schema.tsx]: Update comment 😅

File: playground/src/specs/circular.ts (hunk 2 above)
Comment (magicmatatjahu) [reply playground/src/specs/circular.ts:1]: It's used in playground. If you develop you can choose schema and it's only used to check that circular (very complex) schemas are also supported.

**Review by jonaslagoni — APPROVED** [review]
👍

**Pull request merged**